src/Equations/damaged-elastic/Kernels/Time.cpp

```diff
@@ -703,3 +692,763 @@ void seissol::kernels::Time::flopsTaylorExpansion(long long& nonZeroFlops,
+        sSm.sxz,
+        sBm.sxz,
+        q);
+  }
+}
+
+void seissol::kernels::Time::stressToDofsDynamicRupture(real* dofsStress, const real* dofs) {
+  const real epsInitxx = m_damagedElasticParameters->epsInitxx;
+  const real epsInityy = m_damagedElasticParameters->epsInityy;
+  const real epsInitzz = m_damagedElasticParameters->epsInitzz;
+  const real epsInitxy = m_damagedElasticParameters->epsInitxy;
+  const real epsInityz = m_damagedElasticParameters->epsInityz;
+  const real epsInitzx = m_damagedElasticParameters->epsInitzx;
+
+  for (unsigned int q = 0; q < NUMBER_OF_ALIGNED_BASIS_FUNCTIONS; q++) {
+    dofsStress[0 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        (dofs[0 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] + epsInitxx);
+
+    dofsStress[1 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        (dofs[1 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] + epsInityy);
+
+    dofsStress[2 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        (dofs[2 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] + epsInitzz);
+
+    dofsStress[3 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        (dofs[3 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] + epsInitxy);
+
+    dofsStress[4 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        (dofs[4 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] + epsInityz);
+
+    dofsStress[5 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        (dofs[5 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] + epsInitzx);
+
+    dofsStress[6 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        dofs[6 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q];
+    dofsStress[7 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        dofs[7 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q];
+    dofsStress[8 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        dofs[8 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q];
+    dofsStress[9 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        dofs[9 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q];
+    dofsStress[10 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q] =
+        dofs[10 * NUMBER_OF_ALIGNED_BASIS_FUNCTIONS + q];
+  }
+}
+
+void seissol::kernels::Time::computeNonLinearBaseFrictionLaw(
+    const seissol::dr::ImpedancesAndEta* impAndEta,
+    unsigned ltsFace,
+    const real* qIPlus,
+    real* qStressIPlus,
+    const real* qIMinus,
+    real* qStressIMinus) {
+  using namespace seissol::dr::misc::quantity_indices;
+  using namespace seissol::dr::misc;
+
+  real lambda0P = impAndEta[ltsFace].lambda0P;
+  real mu0P = impAndEta[ltsFace].mu0P;
+  real lambda0M = impAndEta[ltsFace].lambda0M;
+  real mu0M = impAndEta[ltsFace].mu0M;
+
+  const real aB0 = m_damagedElasticParameters->aB0;
+  const real aB1 = m_damagedElasticParameters->aB1;
+  const real aB2 = m_damagedElasticParameters->aB2;
+  const real aB3 = m_damagedElasticParameters->aB3;
+
+  for (unsigned o = 0; o < CONVERGENCE_ORDER; ++o) {
+    for (unsigned i = 0; i < seissol::dr::misc::numPaddedPoints; i++) {
+
+      real EspIp = (qIPlus[o * numQuantities * numPaddedPoints + XX * numPaddedPoints + i]) +
```

Comment:
Please also consider the solution with the lambda as explained above.

Reply:
Modified this with the lambda function

Reply:
Done